ui/src/utils/realtime/implementation.ts

```diff
@@ -0,0 +1,121 @@
+import { neverEver } from '@/utils/never-ever';
+import { createRealtimeApiConnection } from '@/utils/realtime/realtime.signalr';
+import type { HubConnection } from '@microsoft/signalr';
+import { HubConnectionState } from '@microsoft/signalr';
+import type {
+	HubStatusMesage,
+	MessageFromServer,
+	MessageFromApp,
+} from './messages';
+/// <reference types="vite/client" />
+
+const serverMessageHandlers: Array<(message: MessageFromServer) => void> = [];
+
+export const realtimeApiEventTarget = {
+	addServerMessageHandler(handler: (message: MessageFromServer) => void) {
+		serverMessageHandlers.push(handler);
+	},
+	sendToServer(message: MessageFromApp) {
+		handleMessageFromWindow(message);
+	},
+};
+
+const gitHash: string = import.meta.env.VITE_GITHASH ?? 'HEAD';
+
+const { connection, connectionPromise } =
+	createRealtimeApiConnection(setupConnection);
+
+function send(message: MessageFromServer) {
+	for (const handler of serverMessageHandlers) {
+		handler(message);
+	}
+}
+
+void (async function () {
+	await connectionPromise;
+	void send(getHubStateMessage());
+})();
```

Comment:
Oh neat!  This is new syntax for me.  I haven't seen void on the left side like this in JavaScript before.

I looked it up in ChatGPT and got a good explanation of it!

```
In JavaScript, the void operator is used to evaluate an expression and then return undefined. When you see void on the left edge of a statement like in your example, it's typically used to execute a function or an expression for its side effects without caring about its return value.

Reply:
Yeah, it avoids a warning of an "unawaited" Promise, in this case!